src/primitives/line/styled.rs

```diff
@@ -52,6 +59,95 @@ impl<C: PixelColor> StyledPixels<PrimitiveStyle<C>> for Line {
     }
 }
 
+fn draw_dotted_line<TS, C, D>(
+    dot: &TS,
+    dotted_line_points: &DottedLinePoints,
+    dot_style: &PrimitiveStyle<C>,
+    target: &mut D,
+) -> Result<(), D::Error>
+where
+    TS: Transform + StyledDrawable<PrimitiveStyle<C>, Output = ()>,
+    C: PixelColor,
+    D: DrawTarget<Color = TS::Color>,
+{
```

Comment:
Not very important, but this could have one less generic parameter:

```suggestion
fn draw_dotted_line<TS, D>(
    dot: &TS,
    dotted_line_points: &DottedLinePoints,
    dot_style: &PrimitiveStyle<D::Color>,
    target: &mut D,
) -> Result<(), D::Error>
where
    TS: Transform + StyledDrawable<PrimitiveStyle<D::Color>, Color = D::Color, Output = ()>,
    D: DrawTarget,
{
```